File: HotDesk_PRAI/TMS_HotDesk_Prai/Backup/Library/Library.Root/Control/LogBase.vb
Namespace Control
    Public MustInherit Class LogBase
        Inherits System.Web.UI.Page

        Protected _list As Other.GenericCollection(Of [Object].Log)

        Protected Overrides Sub OnInit(ByVal e As System.EventArgs)
            Me.BindKey()
            MyBase.OnInit(e)
            Me.BindData()
        End Sub

        Protected Overrides Sub OnLoad(ByVal e As System.EventArgs)
            MyBase.OnLoad(e)
        End Sub

        Private Sub BindKey()
            For Each _query As String In Request.QueryString
                If Not String.IsNullOrEmpty(_query) Then
                    Select Case _query
                        Case "id"
                            Me._key = Server.UrlDecode(Request.QueryString("id"))
                        Case "key"
                            Me._setupKey = Request.QueryString("key")
                        Case "page"
                            If IsNumeric(Request.QueryString("page")) Then
                                Me._Pageno = Request.QueryString("page")
                            Else
                                Me._Pageno = 1
                            End If
                    End Select
                End If
            Next
        End Sub

        Protected MustOverride Sub BindData()

        Private _key As String = String.Empty
        Public ReadOnly Property Key() As String
            Get
                Return _key
            End Get
        End Property

        Private _Pageno As Integer = 0
        Public Property PageNo() As Integer
            Get
                Return _Pageno
            End Get
            Set(ByVal value As Integer)
                _Pageno = value
            End Set
        End Property

        Private _setupKey As String = String.Empty
        Public ReadOnly Property SetupKey() As String
            Get
                Return _setupKey
            End Get
        End Property

        Public ReadOnly Property KeyDesc() As String
            Get
                If _list.TotalRow > 0 Then
                    Return _list.Data.Item(0).KeyDesc
                Else
                    Return String.Empty
                End If
            End Get
        End Property

        Public MustOverride ReadOnly Property LogTitle() As String
        Public MustOverride ReadOnly Property LogPage() As String

        Public ReadOnly Property NormalTitle() As String
            Get
                Return Me.LogTitle
            End Get
        End Property

        Public ReadOnly Property DisplayTitle() As String
            Get
                Return Me.LogTitle & " Audit Trail"
            End Get
        End Property

        Public ReadOnly Property GenerateList() As String
            Get
                Return ResolveUrl(Me.LogPage & "?id=" & Server.UrlEncode(Me.Key) & "&key=" & Me.SetupKey & "&page=" & Me.PageNo)
            End Get
        End Property
    End Class
End Namespace

File: HotDesk_PRAI/Library/Library.Root/Control/LogBase.cs
namespace Library.Root.Control
{
    public abstract class LogBase : System.Web.UI.Page
    {
        protected Library.Root.Other.GenericCollection<Library.Root.Object.Log> _list;

        protected override void OnInit(System.EventArgs e)
        {
            this.BindKey();
            base.OnInit(e);
            this.BindData();
        }

        protected override void OnLoad(System.EventArgs e)
        {
            base.OnLoad(e);
        }

        private void BindKey()
        {
            foreach (string _query in Request.QueryString)
            {
                if (!string.IsNullOrEmpty(_query))
                {
                    switch (_query)
                    {
                        case "id":
                            this._key = Server.UrlDecode(Request.QueryString["id"]);
                            break;
                        case "key":
                            this._setupKey = Request.QueryString["key"];
                            break;
                        case "page":
                            int pageVal;
                            if (int.TryParse(Request.QueryString["page"], out pageVal))
                            {
                                this._Pageno = pageVal;
                            }
                            else
                            {
                                this._Pageno = 1;
                            }
                            break;
                    }
                }
            }
        }

        protected abstract void BindData();

        private string _key = string.Empty;
        public string Key
        {
            get { return _key; }
        }

        private int _Pageno = 0;
        public int PageNo
        {
            get { return _Pageno; }
            set { _Pageno = value; }
        }

        private string _setupKey = string.Empty;
        public string SetupKey
        {
            get { return _setupKey; }
        }

        public string KeyDesc
        {
            get
            {
                if (_list.TotalRow > 0)
                {
                    return _list.Data[0].KeyDesc;
                }
                else
                {
                    return string.Empty;
                }
            }
        }

        public abstract string LogTitle { get; }
        public abstract string LogPage { get; }

        public string NormalTitle
        {
            get { return this.LogTitle; }
        }

        public string DisplayTitle
        {
            get { return this.LogTitle + " Audit Trail"; }
        }

        public string GenerateList
        {
            get
            {
                return ResolveUrl(this.LogPage + "?id=" + Server.UrlEncode(this.Key) + "&key=" + this.SetupKey + "&page=" + this.PageNo);
            }
        }
    }
}
